### PR TITLE
Ical research

### PR DESCRIPTION
**Pull request by ignatIgnatov**

### CODE REVIEW

File: src/main/java/ludogorie_soft/reservations_platform_api/controller/PropertyController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import ludogorie_soft.reservations_platform_api.dto.PropertyResponseDto;
+import ludogorie_soft.reservations_platform_api.service.PropertyService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.FileNotFoundException;
+
+@RestController
+@RequestMapping("/api/property")
```
Comment (emiliyank) [inline src/main/java/ludogorie_soft/reservations_platform_api/controller/PropertyController.java:14]: better use plural form `/api/properties`
https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5

File: src/main/java/ludogorie_soft/reservations_platform_api/entity/Property.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    @JoinColumn(name = "user_id")
+    private User owner;
+
+    private String name;
+    private String type;
+
+    private String airBnbUrl;
+    private String bookingUrl;
+
+    private String syncUrl;
```
Comment (emiliyank) [inline src/main/java/ludogorie_soft/reservations_platform_api/entity/Property.java:26]: Can we think of even more descriptive name - for example `iCalSyncUrl` or smth else

File: testAirBnbDirectory/airBnbCalendar-1.ics [hunk 3]
```diff
@@ -0,0 +1,19 @@
+BEGIN:VCALENDAR
```
Comment (emiliyank) [inline testAirBnbDirectory/airBnbCalendar-1.ics:1]: Seems that the content of this directory should be part of .gitignore
And the directory should have better name

File: testDirectory/1.ics [hunk 4]
```diff
@@ -0,0 +1,12 @@
+BEGIN:VCALENDAR
```
Comment (emiliyank) [inline testDirectory/1.ics:1]: Seems that the content of this directory should be part of .gitignore
And it should have better name.

File: src/main/resources/calendar/air-bnb/airBnbCalendar-3.ics [hunk 5]
```diff
@@ -0,0 +1,19 @@
+BEGIN:VCALENDAR
```
Comment (emiliyank) [inline src/main/resources/calendar/air-bnb/airBnbCalendar-3.ics:1]: do we need this file?

**Review by emiliyank — CHANGES_REQUESTED** [review]

File: src/main/java/ludogorie_soft/reservations_platform_api/controller/PropertyController.java (hunk 1 above)
Comment (ignatIgnatov) [reply src/main/java/ludogorie_soft/reservations_platform_api/controller/PropertyController.java:14]: Done!

File: src/main/java/ludogorie_soft/reservations_platform_api/entity/Property.java (hunk 2 above)
Comment (ignatIgnatov) [reply src/main/java/ludogorie_soft/reservations_platform_api/entity/Property.java:26]: Done!

File: testAirBnbDirectory/airBnbCalendar-1.ics (hunk 3 above)
Comment (ignatIgnatov) [reply testAirBnbDirectory/airBnbCalendar-1.ics:1]: Done!

File: testDirectory/1.ics (hunk 4 above)
Comment (ignatIgnatov) [reply testDirectory/1.ics:1]: Done!

File: src/main/resources/calendar/air-bnb/airBnbCalendar-3.ics (hunk 5 above)
Comment (ignatIgnatov) [reply src/main/resources/calendar/air-bnb/airBnbCalendar-3.ics:1]: Only for testing via Postman... I will delete calendar directory before the push

File: testDirectory/1.ics (hunk 4 above)
Comment (ignatIgnatov) [reply testDirectory/1.ics:1]: I added logic to delete them in
```
@AfterEavh
void cleanUp()
```

File: src/main/java/ludogorie_soft/reservations_platform_api/service/impl/CalendarServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+    private static final String BASE_URL = "http://localhost:8080/calendar/ical/";
+
+    @Value("${booking.ics.airBnb.directory}")
+    private String icsAirBnbDirectory;
+
+    @Value("${booking.ics.myCal.directory}")
+    private String icsMyCalDirectory;
+
+    public String getMyCalendar(Long propertyId) throws IOException {
```
Comment (emiliyank) [inline src/main/java/ludogorie_soft/reservations_platform_api/service/impl/CalendarServiceImpl.java:45]: Can we think of a better method name here? Seems to me that this method is like `generateCalendarFile()`

Comment (ignatIgnatov) [reply src/main/java/ludogorie_soft/reservations_platform_api/service/impl/CalendarServiceImpl.java:45]: Done!

**Pull request closed without merging**